### ISSUE TITLE
Enhancement: Insert an Event between 2 others (MoM)

### ISSUE THREAD

**Issue by FASTNAVi** [NPBruce/valkyrie#945]
Hi guys,

I have an idea for Enhancement - I am doing scenarios for MoM. But I have no idea how difficult will be to program this. I will try to explain my idea:

I have an event 1. From this tree of events continue to event 2. I will welcome a button, which will insert an event between event 1 and 2. That mean, after using this button, I will have immediatly new tree of events like this:

Before:     EVENT1 -> EVENT2
After:      EVENT1 -> INSERTED EVENT -> EVENT2

or also

Before:     EVENT1 -> EVENT2
                EVENT1 -> EVENT3
                EVENT1 -> EVENT4
After:       EVENT1 -> INSERTED EVENT -> EVENT2
               EVENT1 -> INSERTED EVENT -> EVENT3
               EVENT1 -> INSERTED EVENT -> EVENT4

I don't know if anybody else will use it, but I am sometimes inserting this events in the middle, and when I have almost 1000 events, looking for the right ones in the list takes some time. And I think my scenario will have around 3000 events in total or even more. This feature will help me a lot.

Please, another authors, will you welcome something like this?

Please, programers, how difficult will be for you implement such a thing?

Thank you for your reactions =)

**Comment by scrubbless** [comment]
While it would be helpful, I can't see how this is possible to program.

Also the sort functions should make I easier to search for events. You just need to be consistent with your naming.

**Comment by antontimmermans** [comment]
I agree with scrubbless. I handle the 'insert event' problem with consistent naming.
I will offer some text to the auditor of https://docs.google.com/document/d/1Gi5gTwKkw84Px0Lvk-bJ3Aml_vF3ZoljjB6a8FjWGt8/edit on how I deal with that.

Although I can imagine that with ~3000 events that even consistent naming and a sorted list can give you a headache when you scroll down to find a particular event. A search option in the editor (or <crtl>+F, our autocomplete as you type the name) would then be useful. But I guess that is a different request entirely.

**Comment by FASTNAVi** [comment]
Yes, at least some kind of "finder" will be helpful.

I have really consistent naming. I spent two weeks just to think about and the invent system of naming, because my scenario will be huge with a lot of variaty (and repeating events in different variations). But I find, that is sometimes better make normal chain of events and then add special events which are triggered only sometimes, in really special conditions (for examply I have a running person token and he can choose from several way to run.. and I need to solve many possibilities, for example what if he is running to the room which you don't see yet, what if he enter a room what you already discover, what if you discover a room in the moment, when he is inside and etc.). So I am doing "normal" static scenario first and on the end I will do this special stuff. 

Just recently I finished similar problem in another part of my scenario. It takes me a lot of time - most of it inside text editor, because for me was much quicker and easier to do it manually then directly inside Valkyrie. But you know, when you are adding 30 events somewhere in the middle of chain of other events, it is easy to do mistake.

And so far, I cannot imagine how long I will need to test my scenario.. It will have more than 50 variations, so it will take ages to find and erase all the mistakes.. will see.. it will be fun.. =)

But anyway, thank you for your responses, I will close this thread now.
